app/thermal_management/thermalmgmt.c

```diff
@@ -0,0 +1,237 @@
+
+	for (uint8_t idx = 0; idx < max_adc_sensors; idx++) {
+		LOG_INF("adc ch: %d, for acpi sen: %d",
+			therm_sensor_tbl[idx].adc_ch,
+			therm_sensor_tbl[idx].acpi_loc);
+
+		adc_ch_bits |= BIT(therm_sensor_tbl[idx].adc_ch);
+	}
+
+	LOG_INF("adc ch sensors bit: %x", adc_ch_bits);
+	if (thermal_sensors_init(adc_ch_bits)) {
+		LOG_WRN("Thermal Sensor module init failed!");
+	} else {
+		thermal_initialized = true;
+	}
+
+}
+
+static uint8_t mapped_fan_speed(void)
+{
+	uint8_t cpu_fan_speed;
+
+	switch(cpu_temp) {
+	case 00 ... 19 :
+		cpu_fan_speed = 0;
+		break;
+	case 20 ... 29 :
+		cpu_fan_speed = 20;
+		break;
+	case 30 ... 39:
+		cpu_fan_speed = 30;
+		break;
+	case 40 ... 49:
+		cpu_fan_speed = 50;
+		break;
+	case 50 ... 59:
+		cpu_fan_speed = 75;
+		break;
+	case 60 ... 100:
+	default:
+		cpu_fan_speed = 100;
+		break;
+	}
+
+	return cpu_fan_speed;
+}
+
+static void manage_fan(void)
+{
+	/* EC Self control fan based on CPU thermal info */
+	uint8_t cpu_fan_speed = mapped_fan_speed();
+	LOG_WRN("Fan speed 0x%x", cpu_fan_speed);
+
+	if (fan_duty_cycle[FAN_CPU] != cpu_fan_speed) {
+		fan_duty_cycle[FAN_CPU] = cpu_fan_speed;
+		fan_duty_cycle_change = 1;
+	}
+
+	/* HW/KConfig override takes precedence over every control method
+	 * This is mostly used for PO entry on PO team request
+	 */
+	if (fan_override) {
+		fan_duty_cycle[FAN_CPU] = CONFIG_THERMAL_FAN_OVERRIDE_VALUE;
+		fan_duty_cycle_change = 1;
+	}
+
+	if (fan_duty_cycle_change) {
+		fan_duty_cycle_change = 0;
+
+		for (uint8_t idx = 0; idx < max_fan_dev; idx++) {
+			fan_set_duty_cycle(idx, fan_duty_cycle[idx]);
+		}
+	}
+
+	for (uint8_t idx = 0; idx < max_fan_dev; idx++) {
+		uint16_t rpm;
+
+		fan_read_rpm(idx, &rpm);
+	}
+}
+
+static void manage_thermal_sensors(void)
+{
+	/* Do not attempt to update if no sensors were detected */
+	if (!thermal_initialized) {
+		return;
+	}
+
+	thermal_sensors_update();
+}
+
+K_TIMER_DEFINE(peci_delay_timer, NULL, NULL);
+static bool host_ready;
+
+void peci_start_delay_timer(void)
+{
+	/* start a one-shot timer for prescribed seconds */
+	k_timer_start(&peci_delay_timer, K_SECONDS(CPU_TEMP_ACCESS_DELAY_SEC),
+		      K_NO_WAIT);
+	LOG_WRN("PECI delay timer started");
+	host_ready = true;
+}
+
+static void manage_cpu_thermal(void)
+{
+	int temp, ret;
+
+	/* Manage CPU thermal only in S0 state */
+	if (!peci_initialized || k_timer_remaining_get(&peci_delay_timer) ||
+	    (pwrseq_system_state() != SYSTEM_S0_STATE)) {
+		return;
+	}
+
+	/* Read cpu temperature using peci. */
+	ret = peci_get_temp(&temp);
+	if (ret) {
+		LOG_ERR("Failed to get cpu temperature host_ready: %d, ret-%x", host_ready, ret);
+		temp = CPU_FAIL_CRITICAL_TEMPERATURE;
+	}
+
+	cpu_temp = temp;
+
+	LOG_WRN("%s: Cpu Temp=%d", __func__, temp);
+}
+
+int thermalmgmt_task_init(void)
+{
+	int err;
+
+	init_fans();
+	init_therm_sensors();
+	err = peci_init();
+	if (!err) {
+		peci_initialized = true;
+	} else {
+		LOG_WRN("Fail to initialize PECI");
```

Comment:
do you want to return non zero error value here?

Reply:
Done